### PR TITLE
Cleanups: isResult() more used, readability improved

### PR DESCRIPTION
**Pull request by Quix0r**
This PR is an assortment of cleanups, like adding empty lines or spaces for improving readability. It also makes more use of `DBA::isResult()` in some places.

Others:
- In `src/Worker/OnePoll.php` type-hints for `updateContact` has been added
- In `src/Protocol/Diaspora.php` type-hint `SimpleXMLElement` added
- remaining spaces converted to tabs in `mod/repair_ostatus.php` for indenting code

### CODE REVIEW

File: src/Database/DBA.php [hunk 1]
```diff
@@ -251,22 +255,20 @@ private static function logIndex($query) {
...
 
 	public static function escape($str) {
-		if (self::$connected) {
-			switch (self::$driver) {
-				case 'pdo':
-					return substr(@self::$connection->quote($str, PDO::PARAM_STR), 1, -1);
-				case 'mysqli':
-					return @self::$connection->real_escape_string($str);
-			}
-		} else {
```
Comment (annando) [inline src/Database/DBA.php]: Have you checked whether the removal of this part could make a problem, possibly during the installation phase?

File: src/Database/DBA.php [hunk 2]
```diff
@@ -251,22 +255,20 @@ private static function logIndex($query) {
...
+			case 'mysqli':
+				return @self::$connection->real_escape_string($str);
 		}
 	}
 
 	public static function connected() {
 		$connected = false;
 
 		if (is_null(self::$connection)) {
+			/// @TODO Shouldn't we better prevent this incorrect invocation rather than failing silently?
```
Comment (annando) [inline src/Database/DBA.php]: In no case we should create an error here. We are calling this function for example from the connect function. With raising an error here we couldn't create a database connection anymore at all.

File: src/Model/Item.php [hunk 3]
```diff
@@ -19,6 +19,7 @@
 use Friendica\Model\Contact;
 use Friendica\Model\PermissionSet;
 use Friendica\Model\ItemURI;
+use Friendica\Model\Profile;
```
Comment (annando) [inline src/Model/Item.php]: It isn't in use in this file.

**Comment by annando** [comment]
Obligatory question: How long have you tested this PR on your system and had you checked the error logs during that time?

File: src/Database/DBA.php (hunk 1 above)
Comment (Quix0r) [reply src/Database/DBA.php]: Huh? Maybe a mistake or something strange. 👍

**Comment by Quix0r** [comment]
To answer your question, it is used for a very long time on my instance. It seems I use `PDO` there. So good catch!

File: src/Database/DBA.php (hunk 1 above)
Comment (Quix0r) [reply src/Database/DBA.php]: And good catch!

File: mod/wall_attach.php [hunk 4]
```diff
@@ -45,32 +45,35 @@ function wall_attach_post(App $a) {
...
+				foreach ($_SESSION['remote'] as $v) {
+					if ($v['uid'] == $page_owner_uid) {
 						$contact_id = $v['cid'];
 						break;
 					}
 				}
 			}
-			if($contact_id) {
 
+			if($contact_id > 0) {
```
Comment (MrPetovan) [inline mod/wall_attach.php]: Standards: Please add a space after `if`.

File: src/Database/DBA.php [hunk 5]
```diff
@@ -91,6 +92,7 @@ public static function connect($serveraddr, $user, $pass, $db, $charset = null)
 				self::$connection->setAttribute(PDO::ATTR_EMULATE_PREPARES, false);
 				self::$connected = true;
 			} catch (PDOException $e) {
+				/// @TODO At least log exception, don't ignore it!
```
[the remark this answers is not in the document]
Comment (MrPetovan) [reply src/Database/DBA.php]: Using the App mode flag, it’s now possible to get a fallback behavior for every system using the database. Config is safe as long as it isn’t called in the DBA class for example.

**Comment by Quix0r** [comment]
Gee, totally lost from radar ... Will try it this weekend (not far anymore).

**Comment by Quix0r** [comment]
Rebase completed and conflict fixed.

File: mod/wall_attach.php (hunk 4 above)
Comment (Quix0r) [reply mod/wall_attach.php]: Added a few more in same file.

File: src/Model/Contact.php [hunk 6]
```diff
@@ -1540,7 +1542,15 @@ public static function createFromProbe($uid, $url, $interactive = false, $networ
...
 
-	public static function updateSslPolicy($contact, $new_policy)
+	/**
+	 * @brief Updated contact's SSL policy
+	 *
+	 * @param array  $contact Contact array
+	 * @param string $new_policy New policy, valid: self,full
+	 *
+	 * @return array Contact array with updated values
+	 */
```
Comment (Quix0r) [inline src/Model/Contact.php]: I maintained the comment here @annando @MrPetovan , should this come back?

File: src/Model/GContact.php [hunk 7]
```diff
@@ -915,6 +915,11 @@ public static function updateForUser($uid)
 			intval($uid)
 		);
 
+		if (!DBA::isResult($r)) {
```
Comment (Quix0r) [inline src/Model/GContact.php]: I added this block, @annando @MrPetovan @tobiasd

File: src/Database/DBA.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (MrPetovan) [reply src/Database/DBA.php]: No, it's ok.

**Comment by Quix0r** [comment]
Where are these classes now located?
````
similarity index 100%
rename from src/Network/HTTPException/InternalServerErrorException.php
rename to src/Module/HTTPException/InternalServerErrorException.php
diff --git a/src/Network/HTTPException/LenghtRequiredException.php b/src/Module/HTTPException/LenghtRequiredException.php
````
Locally they are in `src/Module/` now.

**Comment by MrPetovan** [comment]
They're supposed to be in `src/Network`. 🤔

**Pull request merged**